### PR TITLE
UpdateVmEncryptionSettings() needs to be called for VolumeType=Data on Linux VMs

### PR DESCRIPTION
**Pull request by krkhan**
[Internal CR](http://codeflow/Dashboard/review/display/kakhan-12dc3da0f6f142fa9834dd055f1d03f6)

### CODE REVIEW

**Comment by krkhan** [comment]
On-demand run: http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-demand/1180/

**Comment by krkhan** [comment]
Kicked off new on-demand run: http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-demand/1182/

**Comment by krkhan** [comment]
Remerged dev, new on-demand run: http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-demand/1189/

**Comment by krkhan** [comment]
@markcowl ping..

File: src/ResourceManager/Compute/Commands.Compute/Models/AzureDiskEncryptionStatusContext.cs [hunk 1]
```diff
@@ -22,6 +22,7 @@ enum EncryptionStatus
         Encrypted,
         NotEncrypted,
         NotMounted,
+        DecryptionInProgress,
```
Comment (cormacpayne) [inline src/ResourceManager/Compute/Commands.Compute/Models/AzureDiskEncryptionStatusContext.cs]: @krkhan is this change needed for this PR, or is it for a separate change?

**Comment by cormacpayne** [comment]
LGTM

Left one comment for clarification

File: src/ResourceManager/Compute/Commands.Compute/Models/AzureDiskEncryptionStatusContext.cs (hunk 1 above)
Comment (krkhan) [inline src/ResourceManager/Compute/Commands.Compute/Models/AzureDiskEncryptionStatusContext.cs]: @cormacpayne Yes the change is related to encryption disable. I can create a separate PR for this one line but it is needed.
